engine/api/application.go

```diff
@@ -29,12 +30,23 @@ func (api *API) getApplicationsHandler() Handler {
 	return func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
 		vars := mux.Vars(r)
 		projectKey := vars["permProjectKey"]
+		withPermissions := r.FormValue("permission")
 
 		applications, err := application.LoadAll(api.mustDB(), api.Cache, projectKey, getUser(ctx))
 		if err != nil {
 			return sdk.WrapError(err, "getApplicationsHandler> Cannot load applications from db")
 		}
 
+		if strings.ToUpper(withPermissions) == "W" {
+			res := make([]sdk.Application, 0, len(applications))
+			for _, a := range applications {
+				if a.Permission >= 5 {
```

Comment:
use this constant https://github.com/ovh/cds/blob/master/engine/api/permission/permission.go#L11